libcxx/include/__expected/expected.h

```diff
@@ -1354,7 +1354,7 @@ class __expected_void_base {
 
 template <class _Tp, class _Err>
   requires is_void_v<_Tp>
-class expected<_Tp, _Err> : private __expected_void_base<_Err> {
+class [[nodiscard]] expected<_Tp, _Err> : private __expected_void_base<_Err> {
```

Comment:
Since we're adding `nodiscard` we should also review all member functions of this class. Probably all observers can be added and maybe there are other functions that should be done. (I assume the monadic operations will already work correctly.)